### PR TITLE
Update to CUDA 12.4

### PR DESCRIPTION
**Pull request by bdice**
This PR updates pynvjitlink to use CUDA 12.4. This depends on https://github.com/conda-forge/libnvjitlink-feedstock/pull/9.

### CODE REVIEW

**Comment by jakirkham** [comment]
Thanks Bradley! 🙏

Think we also need PR: https://github.com/conda-forge/cuda-nvcc-feedstock/pull/37

**Comment by jakirkham** [comment]
Think we have what we need. Trying to restart CI to see if that fixes the issue

**Comment by jakirkham** [comment]
Looks like the Conda builds completed successfully! 🎉

CI is still running

**Comment by bdice** [comment]
Awesome. After this CI run completes, I want to push one more change: ab2ec91af7588ffcab161053e00e5279806d6fc1 from #60, since it fixes a bug in the wheel testing script.

**Comment by jakirkham** [comment]
Is it worth putting that in its own PR?

Happy to merge that fix first

**Comment by bdice** [comment]
I saw enough of the CI output. I pushed the fix for wheel tests so we are testing the actual CI artifacts rather than nightly packages.

File: ci/test_wheel.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ RAPIDS_TESTS_DIR=${RAPIDS_TESTS_DIR:-"${PWD}/test-results"}/
 mkdir -p "${RAPIDS_TESTS_DIR}"
 
 rapids-logger "Install wheel"
-python -m pip install --find-links ./dist pynvjitlink${PACKAGE_CUDA_SUFFIX}
+python -m pip install $(echo ./dist/pynvjitlink*.whl)
```
Comment (jakirkham) [inline ci/test_wheel.sh:25]: Do we still want the CUDA suffix? Or does that matter?

If we don't need the suffix, should we drop `PACKAGE_CUDA_SUFFIX` from above?

```suggestion
python -m pip install $(echo ./dist/pynvjitlink-${PACKAGE_CUDA_SUFFIX}*.whl)
```

Comment (bdice) [reply ci/test_wheel.sh]: I pushed a change for this.

**Review by jakirkham — APPROVED** [review]
LGTM. Thanks Bradley! 🙏

**Comment by jakirkham** [comment]
Looks like we do this the old school way

**Pull request merged**